build-tools/packages/build-tools/src/common/monoRepo.ts

```diff
@@ -103,63 +104,66 @@ export class MonoRepo {
 
 		this._packageJson = readJsonSync(packagePath);
 
+		const {
+			tool: packageManager,
+			rootDir,
+			packages: discoveredPackages,
+		} = getPackagesSync(repoPath);
 		this.packageManager = existsSync(pnpmWorkspace)
 			? "pnpm"
 			: existsSync(yarnLockPath)
 			? "yarn"
 			: "npm";
+
+		if (this.packageManager !== packageManager.type) {
+			throw new Error(
+				`ERROR: package manager mismatch between ${packageManager.type} and ${this.packageManager}`,
+			);
+		}
+
 		if (existsSync(lernaPath)) {
 			const lerna = readJsonSync(lernaPath);
+			if (this.packageManager === "pnpm") {
+				const workspaceString = readFileSync(pnpmWorkspace, "utf-8");
+				this.workspaceGlobs = YAML.parse(workspaceString).packages;
```

Comment:
Rather than extracting and interpreting the workspace globs, why not just ask pnpm what packages are in the workspace, and store a list of them (instead of the globs)? `pnpm recursive list --json --depth=-1` will give you a nice json formatted list of packages with some info about them.

Reply:
The globs are currently used when we call npm-check-updates to check for new versions: 

https://github.com/microsoft/FluidFramework/blob/579230711e4c10558b0745fd7263a08e9d5db458/build-tools/packages/build-cli/src/lib/package.ts#L107-L113

It's a lot faster that way. We use ncu's API because it already knows how to check npm for updates based on dep ranges etc, and we support updating to prereleases, whose release data is not in the repo.